### PR TITLE
rescrobbled: init at 0.4.0

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Something like mpris-scrobbler but more configurable.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/audio/rescrobbled/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+
+  buildInputs = [
+    dbus
+    openssl
+  ];
+
+  postPatch = ''
+    # Required for tests.
+    substituteInPlace src/filter.rs --replace '#!/usr/bin/bash' '#!${runtimeShell}'
```
Comment (legendofmiracles) [inline pkgs/tools/audio/rescrobbled/default.nix:34]: Using runtimeShell in tests feels a bit awkward.. I don't think it's very important but maybe `stdenv.shell` or just plain `${bash}/bin/bahs`

Comment (urandom2) [reply pkgs/tools/audio/rescrobbled/default.nix:34]: agreed this should be
```suggestion
    substituteInPlace src/filter.rs --replace '#!/usr/bin/bash' '#!${bash}/bin/bash'
```

File: pkgs/tools/audio/rescrobbled/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "rescrobbled";
+  version = "0.4.0";
+
+  src = fetchFromGitHub {
+    owner = "InputUsername";
+    repo = "rescrobbled";
+    rev = "v${version}";
+    sha256 = "stz0mrXEFDGx+XLxMttb2XXR5n7cpw86USZQiUaagSw=";
```
Comment (urandom2) [inline pkgs/tools/audio/rescrobbled/default.nix:18]: nit
```suggestion
    hash = "stz0mrXEFDGx+XLxMttb2XXR5n7cpw86USZQiUaagSw=";
```

File: pkgs/tools/audio/rescrobbled/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  };
+
+  cargoSha256 = "IxlhLIIHvEFBd+PghLsRq4PIxFwnHBJd0+79pxelzi4=";
```
Comment (urandom2) [inline pkgs/tools/audio/rescrobbled/default.nix:21]: nit
```suggestion
  cargoHash = "IxlhLIIHvEFBd+PghLsRq4PIxFwnHBJd0+79pxelzi4=";
```

**Comment by jtojnar** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/274553

**Pull request closed without merging**